platform/ios/src/MGLMapView.mm

```diff
@@ -1264,7 +1265,6 @@ - (void)handleCompassTapGesture:(__unused id)sender
 
 - (void)touchesBegan:(__unused NSSet<UITouch *> *)touches withEvent:(__unused UIEvent *)event
 {
-    _changeDelimiterSuppressionDepth = 0;
```

Comment:
If you perform the following gestures:

1. Swipe the map forcefully enough that it drifts.
1. Before the map finishes drifting, swipe the map in the other direction forcefully enough that it drifts.

Does the map call the delegate methods in the right order?

1. `-mapView:regionWillChangeAnimated:`
1. `-mapViewIsChangingAnimated:` many times
1. `-mapView:regionDidChangeAnimated:`
1. `-mapView:regionWillChangeAnimated:`
1. `-mapViewIsChangingAnimated:` many times
1. `-mapView:regionDidChangeAnimated:`

In particular, do `-mapView:regionWillChangeAnimated:` and `-mapView:regionDidChangeAnimated:` get called only once per swipe? Do subsequent gestures like taps continue to call `-mapView:regionWillChangeAnimated:` and `-mapView:regionDidChangeAnimated:` as expected?

Now that we have an integration test harness, we can write tests for these delegate method calls.

Reply:
@1ec5  I've tested with manual way and the tests from https://github.com/mapbox/mapbox-gl-native/pull/12313, the result is this change will not make the region events unordered. But seems the CI is hanging on some steps. I have no access to rerun it.

Reply:
You’ll need to rebase this PR atop master again. Looks like we’ve been busy adding required builds…

Reply:
@1ec5 Just rebased atop master, the test coverage decreased -1.007% since the first build of this branch.😂